src/main/resources/static/templates/XXEVulnerability/LEVEL_1/XXE.js

```diff
@@ -14,68 +7,55 @@ let buildXmlRequest = function (
 ) {
   let xmlRequestDocument = document.implementation.createDocument("", "", null);
   let bookElement = xmlRequestDocument.createElement("book");
+
   let bookNameElement = xmlRequestDocument.createElement("name");
-  bookNameElement.innerHTML = bookName;
+  bookNameElement.textContent = bookName; // Use textContent to prevent XSS
   bookElement.appendChild(bookNameElement);
 
   let authorElement = xmlRequestDocument.createElement("author");
-  authorElement.innerHTML = author;
+  authorElement.textContent = author; // Use textContent to prevent XSS
   bookElement.appendChild(authorElement);
 
   let isbnElement = xmlRequestDocument.createElement("isbn");
-  isbnElement.innerHTML = isbn;
+  isbnElement.textContent = isbn; // Use textContent to prevent XSS
   bookElement.appendChild(isbnElement);
 
   let publisherElement = xmlRequestDocument.createElement("publisher");
-  publisherElement.innerHTML = publisher;
```

Comment:
CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')
------
User controlled data in methods like `innerHTML`, `outerHTML` or `document.write` is an anti-pattern that can lead to XSS vulnerabilities